sdk/core/System.ClientModel/src/Internal/ContentTypeUtilities.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace System.ClientModel.Internal;
+
+internal class ContentTypeUtilities
+{
+    public static bool TryGetTextEncoding(string contentType, out Encoding? encoding)
+    {
+        const string charsetMarker = "; charset=";
+        const string utf8Charset = "utf-8";
+        const string textContentTypePrefix = "text/";
+        const string jsonSuffix = "json";
+        const string appJsonPrefix = "application/json";
+        const string xmlSuffix = "xml";
+        const string urlEncodedSuffix = "-urlencoded";
+
+        // Default is technically US-ASCII, but will default to UTF-8 which is a superset.
+        const string appFormUrlEncoded = "application/x-www-form-urlencoded";
+
+        if (contentType == null)
+        {
+            encoding = null;
+            return false;
+        }
+
+        var charsetIndex = contentType.IndexOf(charsetMarker, StringComparison.OrdinalIgnoreCase);
```

Comment:
Is there any benefit to using the Span-based operations for `IndexOf`, `StartsWith`, and `EndsWith`?